docs/sources/flow/tutorials/flow-by-example/logs-and-relabeling-basics/index.md

```diff
@@ -0,0 +1,307 @@
+
+Before we move on to logs, let's take a look at how we can use the `prometheus.relabel` component to relabel metrics. The `prometheus.relabel` component allows you to perform Prometheus relabeling on metrics and is similar to the `relabel_configs` section of a Prometheus scrape config.
+
+Let's add a `prometheus.relabel` component to a basic pipeline and see how to add labels.
+
+```river
+prometheus.exporter.unix "localhost" { }
+
+prometheus.scrape "default" {
+    scrape_interval = "10s"
+
+    targets    = prometheus.exporter.unix.localhost.targets
+    forward_to = [
+        prometheus.relabel.example.receiver,
+    ]
+}
+
+prometheus.relabel "example" {
+    forward_to = [
+        prometheus.remote_write.local_prom.receiver,
+    ]
+
+    rule {
+        action       = "replace"
+        target_label = "os"
+        replacement  = constants.os
+    }
+}
+
+prometheus.remote_write "local_prom" {
+    endpoint {
+        url = "http://localhost:9090/api/v1/write"
+    }
+}
+```
+
+We have now created the following pipeline:
+
+<p align="center">
+<img src="/media/docs/agent/diagram-flow-by-example-relabel-0.svg" alt="Diagram of pipeline that scrapes prometheus.exporter.unix, relabels the metrics, and remote_writes them." width="500" />
+</p>
+
+This pipeline has a `prometheus.relabel` component that has a single rule. This rule has the `replace` action, which will replace the value of the `os` label with a special value: `constants.os`. This value is a special constant that will be replaced with the os of the host the Agent is running on. You can see the other available constants in the [constants](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/stdlib/constants/) documentation page. In this example we have one rule block, but you can have as many as you want. Each rule block will be applied in order.
+
+If you run the Agent and navigate to [localhost:3000/explore](http://localhost:3000/explore), you should be able to see the `os` label on the metrics. Try querying for `node_context_switches_total` and look at the labels!
+
+Relabeling uses the same rules as Prometheus. You can always refer to the [prometheus.relabel documentation](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/prometheus.relabel/#rule-block) for a full list of available options.
+
+{{% admonition type="note" %}}
+You can have multiple components forward to one `prometheus.relabel` component. This allows you to apply the same relabeling rules to multiple pipelines.
+{{% /admonition %}}
+
+{{% admonition type="warning" %}}
+There is a common "gotcha" when using metrics that start with `__` (double underscore). These labels are considered internal and will be dropped before relabeling rules from a `prometheus.relabel` component are applied. If you would like to keep or act on these kinds of labels, use a [discovery.relabel](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/discovery.relabel/) component.
```

Comment:
I have been thinking about this one... and might have some better wording.

Reply:
```suggestion
When metric labels start with `__` (double underscore), they are considered internal labels and are dropped before the relabeling rules from a `prometheus.relabel` component are applied. If you would like to keep or act on these labels, use a [discovery.relabel](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/discovery.relabel/) component.
```

Maybe this is better?

Reply:
This caught a mistake on my part: that the issue is _labels_ that start with `__`, not metrics. However, it's not inherently an issue with the label starting with `__` (this is part of Prometheus) but rather an issue commonly faced by users trying to use or keep those labels, since they have already been dropped.

How does:

```suggestion
There is an issue commonly faced when relabeling and using labels that start with `__` (double underscore). These labels are considered internal and are dropped before relabeling rules from a `prometheus.relabel` component are applied. If you would like to keep or act on these kinds of labels, use a [discovery.relabel](https://grafana.com/docs/agent/<AGENT_VERSION>/flow/reference/components/discovery.relabel/) component.
```
sound?

Reply:
Yeah that works... and makes more sense now :-)